src/transformers/models/mask2former/image_processing_mask2former.py

```diff
@@ -1004,55 +1008,108 @@ def post_process_instance_segmentation(
             labels_per_image = labels[topk_indices]
 
             topk_indices = torch.div(topk_indices, num_classes, rounding_mode="floor")
             mask_pred = mask_pred[topk_indices]
-            pred_masks = (mask_pred > 0).float()
 
+            segmentation = torch.zeros(mask_size) - 1
+
+            pred_classes = labels_per_image
+
+            # calculate final prediction scores and masks
+            pred_masks = (mask_pred > 0).float()
             # Calculate average mask prob
             mask_scores_per_image = (mask_pred.sigmoid().flatten(1) * pred_masks.flatten(1)).sum(1) / (
                 pred_masks.flatten(1).sum(1) + 1e-6
             )
             pred_scores = scores_per_image * mask_scores_per_image
-            pred_classes = labels_per_image
 
-            segmentation = torch.zeros((384, 384)) - 1
+            # resize prediction masks
             if target_sizes is not None:
                 segmentation = torch.zeros(target_sizes[i]) - 1
                 pred_masks = torch.nn.functional.interpolate(
                     pred_masks.unsqueeze(0), size=target_sizes[i], mode="nearest"
                 )[0]
 
-            instance_maps, segments = [], []
-            current_segment_id = 0
-            for j in range(num_queries):
-                score = pred_scores[j].item()
-
-                if not torch.all(pred_masks[j] == 0) and score >= threshold:
-                    segmentation[pred_masks[j] == 1] = current_segment_id
-                    segments.append(
-                        {
-                            "id": current_segment_id,
-                            "label_id": pred_classes[j].item(),
-                            "was_fused": False,
-                            "score": round(score, 6),
-                        }
-                    )
-                    current_segment_id += 1
-                    instance_maps.append(pred_masks[j])
-                    # Return segmentation map in run-length encoding (RLE) format
-                    if return_coco_annotation:
-                        segmentation = convert_segmentation_to_rle(segmentation)
-
-            # Return a concatenated tensor of binary instance maps
-            if return_binary_maps and len(instance_maps) != 0:
-                segmentation = torch.stack(instance_maps, dim=0)
+            # Compute segmentation maps and segments_info
+            segmentation, segments = self._post_process_instance_segmentation(
+                num_topk_queries,
+                pred_masks,
+                pred_classes,
+                pred_scores,
+                segmentation,
+                threshold,
+                return_coco_annotation,
+                return_binary_maps,
+            )
 
             results.append({"segmentation": segmentation, "segments_info": segments})
+
         return results
 
+    def _post_process_instance_segmentation(
+        self,
+        num_predictions: int,
+        pred_masks: torch.Tensor,
```

Comment:
The currently failing tests are coming from the typing here. All the torch objects in typing should be converted to strings. 
```suggestion
        pred_masks: "torch.Tensor",
```

Reply:
oh okay got it! Thanks for the help @amyeroberts :)